src/ic-cdk/src/api/call.rs

```diff
@@ -404,7 +407,16 @@ pub fn call<T: ArgumentEncoder, R: for<'a> ArgumentDecoder<'a>>(
     let fut = call_raw(id, method, &args_raw, 0);
     async {
         let bytes = fut.await?;
-        decode_args(&bytes).map_err(|err| trap(&format!("{:?}", err)))
+        decode_args(&bytes).map_err(|err| {
```

Comment:
`call_with_payment` and `call_with_payment128` below need the same fix.

Reply:
Good catch, done!